kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -387,7 +404,7 @@ class ResidualBasedBlockBuilderAndSolver
             BaseType::mpLinearSystemSolver->Solve(A, Dx, b);
         } else {
             TSparseSpace::SetToZero(Dx);
-            KRATOS_WARNING("ResidualBasedBlockBuilderAndSolver") << "ATTENTION! setting the RHS to zero!" << std::endl;
+            KRATOS_WARNING_IF("ResidualBasedBlockBuilderAndSolver", mOptions.IsNot(SILENT_WARNINGS)) << "ATTENTION! setting the RHS to zero!" << std::endl;

```

Comment:
@RiccardoRossi what do you think abt this?
This warning annoys me too (in the tests) but in real cases it is very important / useful

Reply:
I am ok with having the possibility of silencing this warning however:
1 - why mixing this feature in this PR?
2 - i would have it on by default, and allow one to switch it to off upon request

Reply:
> 1 - why mixing this feature in this PR?

+1 for making a separate PR

> 2 - i would have it on by default, and allow one to switch it to off upon request

+1
For us this is probably THE most useful warning to tell if sth is not converging